### PR TITLE
Prevent leaving the page when the update is in progress

### PR DESCRIPTION
**Pull request by Quetzacoalt91**
| Questions         | Answers
| ------------------| -------------------------------------------------------
| Description?      | To avoid accidents while the update is in progress, we had the confirmation modal when the users attempts to leave the page.
| Type?             | improvement
| BC breaks?        | Nope
| Deprecations?     | Nope
| Fixed ticket?     | /
| Sponsor company   | @PrestaShopCorp
| How to test?      | While the update is in progress, try refreshing or closing the page. You will be asked to confirm the action:

Capture d’écran du 2024-12-12 14-59-21

### CODE REVIEW

File: _dev/src/ts/pages/UpdatePageUpdate.ts
Comment (M0rgan01) [inline _dev/src/ts/pages/UpdatePageUpdate.ts]:
```suggestion
  #enableExitConfirmation = (): void => {
    window.addEventListener('beforeunload', this.#handleBeforeUnload);
  };

  #disableExitConfirmation = (): void => {
    window.removeEventListener('beforeunload', this.#handleBeforeUnload);
  };

  #handleBeforeUnload = (event: Event): void => {
    event.preventDefault();

    // Included for legacy support, e.g. Chrome/Edge < 119
    event.returnValue = true;
  };
```

File: _dev/src/ts/pages/UpdatePageUpdate.ts
Comment (ga-devfront) [inline _dev/src/ts/pages/UpdatePageUpdate.ts:127]: what do you think to export this function inside new `domUtils` to be reusable on other script without duplication ? Or add the methods inside PageAbstract ?

Comment (M0rgan01) [reply _dev/src/ts/pages/UpdatePageUpdate.ts:127]: This will actually be useful in the backup and restore processes. +1

Comment (Quetzacoalt91) [reply _dev/src/ts/pages/UpdatePageUpdate.ts:127]: There is already a topic about duplicated code started in https://github.com/PrestaShop/autoupgrade/pull/1076, the objective here was to unblock your PR.

**Review by ga-devfront — APPROVED** [review]

**Comment by ps-jarvis** [comment]
PR merged, well done!

Message to @PrestaShop/committers: do not forget to milestone it before the merge.

**Pull request merged**
